federatedscope/cl/dataloader/Cifar10.py

```diff
@@ -0,0 +1,154 @@
+import math
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.backends.cudnn as cudnn
+import torchvision
+import torchvision.transforms as T
+import torchvision.transforms.functional as TF
+from torch.utils.data import DataLoader, Dataset
+from torchvision.datasets import CIFAR10, CIFAR100
+import pickle as pkl
+import numpy as np
+from federatedscope.register import register_data
+from federatedscope.core.auxiliaries.splitter_builder import get_splitter
+
+
+
+class SimCLRTransform():
+    def __init__(self, is_sup, image_size=32):
+        self.transform = T.Compose([
+            T.RandomResizedCrop(image_size, scale=(0.5, 1.0), interpolation=T.InterpolationMode.BICUBIC),
+            T.RandomHorizontalFlip(p=0.5),
+            T.RandomApply([T.ColorJitter(0.4,0.4,0.2,0.1)], p=0.8),
+            T.RandomGrayscale(p=0.2),
+            T.RandomApply([T.GaussianBlur(kernel_size=3, sigma=(0.1, 2.0))], p=0.5),
+            T.ToTensor(),
+            T.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
+        ])
+
+        self.mode = is_sup
+
+    def __call__(self, x):
+        if(self.mode):
+            return self.transform(x)
+        else:
+            x1 = self.transform(x)
+            x2 = self.transform(x)
+            return x1, x2 
+
+def Cifar4CL(config):
+    
+    transform_train = SimCLRTransform(is_sup=False, image_size=32)
+
+    path = config.data.root
+    
+    data_train = CIFAR10(path, train=True, download=True, transform=transform_train)
+    data_test = CIFAR10(path, train=False, download=True, transform=transform_train)
+    
+    # Split data into dict
+    data_dict = dict()
+    splitter = get_splitter(config)
+    data_train = splitter(data_train)
```

Comment:
splitting `data_train` and `data_test` respectively cannot ensure the iid property for the train and test of the same client.

Reply:
@rayrayraykk  any suggestion to modify the splitter to enable splitting (data1, data2, ...) with the same categorical distributions?

Reply:
We have the args to control the dist.
See https://github.com/alibaba/FederatedScope/blob/1fe2880de4174f9c010178f273756ec6392dedcc/federatedscope/core/auxiliaries/data_builder.py#L583 .

Reply:
add a paramater to decide using the same splitter of training and evaluation or not.

Reply:
keep the train val test splits iid: https://github.com/alibaba/FederatedScope/blob/ccafa143ce6c1f7b18dca1a308e15ac2295e69ff/federatedscope/core/splitters/generic/lda_splitter.py#L11